Frameworks/CoreText/CTFramesetter.mm

```diff
@@ -27,15 +27,51 @@ @implementation _CTFramesetter : NSObject
 @end
 
 static _CTFrame* __CreateFrame(_CTFramesetter* framesetter, CGRect frameSize, CFRange range) {
-    RETURN_NULL_IF(!framesetter);
+    RETURN_NULL_IF(framesetter == nil);
 
     // Call _DWriteWrapper to get _CTLine object list that makes up this frame
     _CTTypesetter* typesetter = static_cast<_CTTypesetter*>(framesetter->_typesetter);
     if (range.length == 0L) {
         range.length = typesetter->_characters.size();
     }
 
-    return [_DWriteGetFrame(static_cast<CFAttributedStringRef>(typesetter->_attributedString.get()), range, frameSize) retain];
+    _CTFrame* ret = _DWriteGetFrame(static_cast<CFAttributedStringRef>(typesetter->_attributedString.get()), range, frameSize);
```

Comment:
>_CTFrame [](start = 4, length = 8)

If you do a `StrongId<_CTFrame>` and then `.detach()` when you return, this will memory manage sanely.